casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala

```diff
@@ -313,30 +313,32 @@ class MultiParentCasperAddBlockSpec extends FlatSpec with Matchers with Inspecto
         _ <- nodes(2).receive()
         _ <- nodes(1).transportLayerEff.clear(nodes(1).local) //nodes(1) misses this block
 
-        _ <- nodes(1).casperEff.contains(signedBlock1) shouldBeF true
-        _ <- nodes(2).casperEff.contains(signedBlock1) shouldBeF false
+        _ <- nodes(1).casperEff.contains(signedBlock1.blockHash) shouldBeF true
+        _ <- nodes(2).casperEff.contains(signedBlock1.blockHash) shouldBeF false
 
-        _ <- nodes(1).casperEff.contains(signedBlock1Prime) shouldBeF false
-        _ <- nodes(2).casperEff.contains(signedBlock1Prime) shouldBeF true
+        _ <- nodes(1).casperEff.contains(signedBlock1Prime.blockHash) shouldBeF false
+        _ <- nodes(2).casperEff.contains(signedBlock1Prime.blockHash) shouldBeF true
 
         signedBlock2 <- nodes(1).addBlock(deployDatas(2))
         signedBlock3 <- nodes(2).addBlock(deployDatas(3))
         _            <- nodes(2).transportLayerEff.clear(nodes(2).local) //nodes(2) ignores block2
-        _            <- nodes(1).receive() // receives block3; asks for block1'
-        _            <- nodes(2).receive() // receives request for block1'; sends block1'
+        _            <- nodes(1).receive() // receives block3; asks if has block1'
+        _            <- nodes(2).receive() // receives request has block1'; sends i have block1'
+        _            <- nodes(1).receive() // receives has block1 ack; asks for block1'
+        _            <- nodes(2).receive() // receives request block1'; sends block1'
         _            <- nodes(1).receive() // receives block1'; adds both block3 and block1'
```

Comment:
Ideally, we'd have a method `exchangeTillSilence(nodes)` that would call receive on all passed nodes in round-robin fashion until all of them have empty message queues. It could have a default iterations threshold/we could lean on the test timeout for cases where silence doesn't happen.

Less ideall, but another option would be to exchange a `requestAndReceiveBlock(node1, node2) that would cover the 4 or 5 receives above, along with their comments

Reply:
ideally we dont want to have tests that look like above

Reply:
Baby steps, my mate, baby steps :joy:

Had a broader look and I admit the .clear usage seem quite hacky. Re: validateBlockStore part - this will be simplified once you rebase on dev :)